### PR TITLE
refactor(storage): refactor row_serde and storage table to match row-based encoding

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As cell-based encoding has been removed, several refactor can be done:
- remove sentinel cell
- refactor interfaces of `row_serde` trait
- refactor `StorageTable` write and read

Next PR will:
- put the `row_serde` on top of the MemTable
- remove `StorageTable` in streaming mode

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by wcy-fdu** [comment]
codecov is not happy

File: src/storage/src/table/storage_table.rs [hunk 1]
```diff
@@ -401,82 +396,24 @@ impl<S: StateStore, RS: RowSerde, const T: AccessType> StorageTableBase<S, RS, T
...
-            deserializer.deserialize(&key, &value).map_err(err)?;
+            .get(&serialized_pk, read_options.clone())
+            .await?
+        {
+            let deserialize_res = deserializer
+                .deserialize(&serialized_pk, &value)
+                .map_err(err)?;
+            match deserialize_res {
+                Some(deserialize_res) => Ok(Some(deserialize_res.2)),
+                None => Ok(None),
```
Comment (BugenZhao) [inline src/storage/src/table/storage_table.rs:412]: When will it be `None`?

File: src/storage/src/table/storage_table.rs [hunk 2]
```diff
@@ -559,13 +495,12 @@ impl<S: StateStore, RS: RowSerde> StorageTableBase<S, RS, READ_WRITE> {
...
-                    let bytes = self
+                    let (key, _) = self
                         .row_serializer
                         .serialize(vnode, &pk, old_row)
                         .map_err(err)?;
-                    for (key, _) in bytes {
-                        local.delete(key);
-                    }
+
+                    local.delete(key);
```
Comment (BugenZhao) [inline src/storage/src/table/storage_table.rs:503]: Can we just use `pk` to delete it?

File: src/storage/src/table/storage_table.rs [hunk 3]
```diff
@@ -401,82 +396,24 @@ impl<S: StateStore, RS: RowSerde, const T: AccessType> StorageTableBase<S, RS, T
...
+                .map_err(err)?;
+            match deserialize_res {
+                Some(deserialize_res) => Ok(Some(deserialize_res.2)),
+                None => Ok(None),
+            }
+        } else {
+            Ok(None)
         }
-
-        let result = deserializer.take();
```
Comment (BugenZhao) [inline src/storage/src/table/storage_table.rs:475]: The `take` interface can be removed too.

File: src/storage/src/table/storage_table.rs (hunk 2 above)
Comment (wcy-fdu) [reply src/storage/src/table/storage_table.rs:503]: So it is, fixed.

File: src/storage/src/table/storage_table.rs [hunk 4]
```diff
@@ -406,11 +406,9 @@ impl<S: StateStore, RS: RowSerde, const T: AccessType> StorageTableBase<S, RS, T
         {
             let deserialize_res = deserializer
                 .deserialize(&serialized_pk, &value)
-                .map_err(err)?;
-            match deserialize_res {
-                Some(deserialize_res) => Ok(Some(deserialize_res.2)),
-                None => Ok(None),
-            }
+                .map_err(err)?
+                .unwrap();
```
Comment (BugenZhao) [inline src/storage/src/table/storage_table.rs:410]: Seems no need to return `Some` and always unwrap here.

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**